ui/src/app/cohort-search/search-group-list/search-group-list.component.tsx

```diff
@@ -167,38 +182,22 @@ const SearchGroupList = fp.flow(withCurrentWorkspace(), withCdrVersions())(
 
     getMenuOptions() {
       this.setState({loadingMenuOptions: true});
-      const {workspace, cdrVersionListResponse} = this.props;
+      const {workspace} = this.props;
       const {cdrVersionId} = workspace;
       const criteriaMenuOptions = criteriaMenuOptionsStore.getValue();
-      cohortBuilderApi().findCriteriaMenuOptions(+cdrVersionId).then(res => {
-        criteriaMenuOptions[cdrVersionId] = res.items.reduce((acc, opt) => {
-          const {domain, types} = opt;
-          if (PROGRAM_TYPES.includes(Domain[domain]) &&
-          !(!getCdrVersion(workspace, cdrVersionListResponse).hasFitbitData && domain === Domain.FITBIT.toString())) {
-            const option = {
-              name: domainToTitle(domain),
-              domain,
-              type: types[0].type,
-              standard: types[0].standardFlags[0].standard,
-              order: PROGRAM_TYPES.indexOf(Domain[domain])
-            };
-            if (domain === Domain[Domain.PERSON]) {
-              option['children'] = types.map(subopt => ({name: typeToTitle(subopt.type), domain, type: subopt.type}));
-            }
-            acc.programTypes.push(option);
-          }
-          if (DOMAIN_TYPES.includes(Domain[domain])) {
-            acc.domainTypes.push({
-              name: domainToTitle(domain),
-              domain,
-              type: types[0].type,
-              standard: types[0].standardFlags[0].standard,
-              order: DOMAIN_TYPES.indexOf(Domain[domain])});
+      cohortBuilderApi().findCriteriaMenu(+cdrVersionId, 0).then(async res => {
+        const menuOptions = await Promise.all(res.items.map(async item => {
+          const option = mapMenuItem(item);
+          if (option.group) {
+            const children = await cohortBuilderApi().findCriteriaMenu(+cdrVersionId, option.id);
+            option.children = children.items.map(mapMenuItem).sort(sortMenuItems);
           }
-          return acc;
-        }, {programTypes: [], domainTypes: []});
-        criteriaMenuOptions[cdrVersionId].programTypes.sort((a, b) => a.order - b.order);
-        criteriaMenuOptions[cdrVersionId].domainTypes.sort((a, b) => a.order - b.order);
+          return option;
+        }));
+        criteriaMenuOptions[cdrVersionId] = {
+          programTypes: menuOptions.filter(opt => opt.category === 'Program Data').sort(sortMenuItems),
+          domainTypes: menuOptions.filter(opt => opt.category === 'Domains').sort(sortMenuItems)
+        };
```

Comment:
The API call does the sort logic in the database call. The list should always be sorted in the proper order in the response.

Reply:
Ok, thanks. Removed the frontend sorting.